HeterogeneousCore/AlpakaTest/plugins/alpaka/TestAlpakaTranscriber.cc

```diff
@@ -48,7 +48,7 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
       portabletest::TestDeviceCollection const& deviceProduct = event.get(deviceToken_);
 
       portabletest::TestHostCollection hostProduct{deviceProduct->metadata().size(), alpaka_common::host(), *device_};
-      alpaka::memcpy(queue, hostProduct.buffer(), deviceProduct.buffer());
+      alpaka::memcpy(queue, hostProduct.buffer(), deviceProduct.const_buffer());
```

Comment:
@makortel this change is not strictly necessary, because `buffer()` also returns a `ConstBuffer` since the `deviceProduct` is not mutable. I've made it explicit here to better clarify the intent.

@tonydp03 and I also checked that swapping the arguments, _e.g._ doing
```c++
alpaka::memcpy(queue, deviceProduct.buffer(), hostProduct.buffer());
```
now results in a compiler error:
```
/data/user/fwyzard/CMSSW_12_5_X_2022-07-11-1100/src/HeterogeneousCore/AlpakaTest/plugins/alpaka/TestAlpakaTranscriber.cc:51:73:   required from here
/cvmfs/cms-ib.cern.ch/nweek-02741/el8_amd64_gcc10/external/alpaka/develop-20220621-4e96939afa0cdb62448c73ead2bb07e0/include/alpaka/mem/view/Traits.hpp:210:23: error: static assertion failed: The destination view's element type must not be const!
  210 |         static_assert(!std::is_const_v<DstElem>, "The destination view's element type must not be const!");
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~
```